src/aggregate/aggregate_exec.c

```diff
@@ -630,6 +630,8 @@ static void sendChunk_Resp3(AREQ *req, RedisModule_Reply *reply, size_t limit,
     } else if (rc == RS_RESULT_ERROR) {
       // Non-fatal error
       RedisModule_Reply_SimpleString(reply, QueryError_GetUserError(req->qiter.err));
+    } else if (req->sctx->spec && req->sctx->spec->scan_failed_OOM) {
+    RedisModule_Reply_SimpleString(reply, QUERY_WINDEXING_FAILURE);
```

Comment:
Are these options mutually exclusive? I believe there can be a query in which we will have both `QUERY_WMAXPREFIXEXPANSIONS` and `QUERY_WMAXPREFIXEXPANSIONS`